### PR TITLE
Merges UI for global search and find in page functionality.

### LINKED ISSUE #154: Try to merge global search + find in page
**Problem**
When in an article (Glaucoma), I wanted to know the specific symptoms of the disease. I went directly to the magnifying glass to search. The magnifying glass is the universal icon for searching.

Because this is really a “global search” and not a search of the contents of the article, my search did not turn up what I was looking for. Instead, I saw a list of other articles. 

The “find in page” feature is important for people looking to find specific content quickly. Since it is behind the 3 dot menu, it was difficult to discover. Though I found it eventually, at first I thought the search didn’t work, and I would have to scan through the Glaucoma article to find the symptoms. 

**Heuristics violated:**

Recognition rather than recall
Minimize the user's memory load by making objects, actions, and options visible. The user should not have to remember information from one part of the dialogue to another. Instructions for use of the system should be visible or easily retrievable whenever appropriate.

Consistency and standards
Users should not have to wonder whether different words, situations, or actions mean the same thing. Follow platform conventions.

Error prevention
Even better than good error messages is a careful design which prevents a problem from occurring in the first place. Either eliminate error-prone conditions or check for them and present users with a confirmation option before they commit to the action.

Compatibility between different platforms (mobile)
Due to fragmentation and constant change in mobile device market, apps must be flexible to adapt to different platforms and devices. The site or platform should automatically detect the kind of device and direct the user to the mobile version. 

Little human/device interaction (mobile)
Due to small size of mobile devices and the context of use (in motion, sometimes with one hand), it is important that the interaction effort is reduced. (see more detail on page 8)

**Recommendation:** 
If possible, make the magnifying glass both universal and local search. Or properly label the search as “search for articles”. Direct users to in article search from there. If this is not possible, find a way to make the local search more immediately discoverable. 

Look into native Android in-article search. People might expect it. It might be buried in OS menus, but perhaps it could be surfaced?

### PR DESCRIPTION
**Pull request by sakchhams**
Fixes #154 

Changes: 
Global Search In-Page

The text in the Global search UI box is carried forward to the in-page search. Also migrates some code using older deprecated APIs to use newer ones where possible.

### CODE REVIEW

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 1]
```diff
@@ -111,7 +113,7 @@ public void finish() {
   public boolean onCreateOptionsMenu(Menu menu) {
     getMenuInflater().inflate(R.menu.menu_search, menu);
     MenuItem searchMenuItem = menu.findItem(R.id.menu_search);
-    MenuItemCompat.expandActionView(searchMenuItem);
+    searchMenuItem.expandActionView();//replaced deprecated method
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Please remove the comment.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 2]
```diff
@@ -132,19 +134,19 @@ public boolean onQueryTextChange(String s) {
       }
     });
 
-    MenuItemCompat.setOnActionExpandListener(searchMenuItem,
-        new MenuItemCompat.OnActionExpandListener() {
-          @Override
-          public boolean onMenuItemActionExpand(MenuItem item) {
-            return false;
-          }
+    //replaced deprecated method
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Here also.

File: app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java [hunk 3]
```diff
@@ -132,19 +134,19 @@ public boolean onQueryTextChange(String s) {
...
+    searchMenuItem.setOnActionExpandListener(new MenuItem.OnActionExpandListener() {
```
Comment (srv-twry) [inline app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Convert to lambda if possible.

Comment (sakchhams) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Sorry. The above can't be converted to a lambda as It overrides more than one method.

Comment (srv-twry) [reply app/src/main/java/org/kiwix/kiwixmobile/search/SearchActivity.java]: Okay :smile:

**Comment by srv-twry** [comment]
Please resolve conflicts.

**Comment by sakchhams** [comment]
@srv-twry conflicts resolved.

**Comment by sakchhams** [comment]
@mhutti1 fixed the whitespace issue.

**Pull request merged**